Digitisers/DCHDigi/src/DCHDigiAlg.cpp

```diff
@@ -29,39 +29,83 @@ DCHDigiAlg::DCHDigiAlg(const std::string& name, ISvcLocator* svcLoc)
   // Output collections
   declareProperty("DigiDCHitCollection", w_DigiDCHCol, "Handle of Digi DCHit collection");
   
-  declareProperty("CaloAssociationCollection", w_AssociationCol, "Handle of Association collection");
+  declareProperty("AssociationCollection", w_AssociationCol, "Handle of Association collection");
    
 }
 
 StatusCode DCHDigiAlg::initialize()
 {
-  /*
   m_geosvc = service<IGeomSvc>("GeomSvc");
   if ( !m_geosvc )  throw "DCHDigiAlg :Failed to find GeomSvc ...";
   dd4hep::Detector* m_dd4hep = m_geosvc->lcdd();
   if ( !m_dd4hep )  throw "DCHDigiAlg :Failed to get dd4hep::Detector ...";
-  m_cellIDConverter = new dd4hep::rec::CellIDPositionConverter(*m_dd4hep);
-  */
+  dd4hep::Readout readout = m_dd4hep->readout(m_readout_name);
+  m_segmentation = dynamic_cast<dd4hep::DDSegmentation::GridDriftChamber*>(readout.segmentation().segmentation());
+  m_decoder = m_geosvc->getDecoder(m_readout_name);
+  if (!m_decoder) {
+      error() << "Failed to get the decoder. " << endmsg;
+      return StatusCode::FAILURE;
+  }
+
+  std::string s_output=m_Output; 
+  if(m_WriteAna){
+      m_fout = new TFile(s_output.c_str(),"RECREATE"); 
```

Comment:
Hi @wenxingfang , 
Could you use Gaudi's NtupleSvc?
Tao

Reply:
Hi @mirguest 
I want to use std::vector<> for branch variable. Does Gaudi's NtupleSvc has similar function ?

Best,
Wenxing

Reply:
Sure, please have a look at https://gitlab.cern.ch/gaudi/Gaudi/-/tree/master/GaudiExamples/src/NTuples
Tao

Reply:
Hi @mirguest Thanks! I have updated it, please check it!

Best,
Wenxing